java-manta-client-unshaded/src/main/java/com/joyent/manta/util/AutoContinuingInputStream.java

```diff
@@ -54,9 +58,16 @@ public AutoContinuingInputStream(final InputStream wrapped,
      */
     private void attemptRecovery(final IOException originalIOException) throws IOException {
         try {
-            this.continueWith(this.continuator.buildContinuation(originalIOException, this.getBytesRead()));
-        } catch (final IOException ce) {
-            originalIOException.addSuppressed(ce);
+            super.continueWith(this.continuator.buildContinuation(originalIOException, this.getBytesRead()));
+        } catch (final IOException ioe) {
```

Comment:
This might be paranoid, but you may want to catch `UncheckedIOException` as well.